src/extension.ts

```diff
@@ -529,14 +530,22 @@ function maybeRunKubernetesCommandForActiveWindow(command, progressMessage) {
                         vscode.window.showErrorMessage("Save failed.");
                         return;
                     }
-                    kubectl.invokeWithProgress(`${command} -f "${editor.document.fileName}"`, progressMessage, resultHandler);
+                    if (editor.document.uri.scheme === 'file') {
+                        kubectl.invokeWithProgress(`${command} -f "${editor.document.fileName}"`, progressMessage, resultHandler);
+                    } else {
+                        kubectlViaTempFile(command, editor.document.getText(), progressMessage, resultHandler);
+                    }
                 });
             }
         });
     } else {
-        const fullCommand = `${command} -f "${editor.document.fileName}"`;
-        console.log(fullCommand);
-        kubectl.invokeWithProgress(fullCommand, progressMessage, resultHandler);
+        if (editor.document.uri.scheme === 'file') {
```

Comment:
Can we extract this as a function and then remove this code duplication by just calling it in both places?

Reply:
Alternately, if we convert the promise into an await we might not even need to extract the code...

Reply:
I'll take the easy route for now but I think you're right - converting this function to async/await would probably simplify the flow a lot.